test/interp1d_test.cu

```diff
@@ -0,0 +1,146 @@
+#include <iostream>
+#include <iomanip>
+#include <math.h>
+#include <helper_cuda.h>
+#include <complex>
+#include "../src/cuspreadinterp.h"
+#include "../contrib/utils.h"
+
+using namespace std;
+
+int main(int argc, char* argv[])
+{
+	int nf1;
+	FLT upsampfac=2.0;
+	int N1, M;
+	if (argc<4) {
+		fprintf(stderr,
+			"Usage: interp1d method nupts_distr nf1 [M [tol [kerevalmeth [sort]]]]\n"
+			"Arguments:\n"
+			"  method: One of\n"
+			"    1: nupts driven\n"
+			"  nupts_distr: The distribution of the points; one of\n"
+			"    0: uniform, or\n"
+			"    1: concentrated in a small region.\n"
+			"  nf1: The size of the 2D array.\n"
+			"  M: The number of non-uniform points (default nf1 / 2).\n"
+			"  tol: NUFFT tolerance (default 1e-6).\n"
+			"  kerevalmeth: Kernel evaluation method; one of\n"
+			"     0: Exponential of square root (default), or\n"
+			"     1: Horner evaluation.\n"
+			"  sort: One of\n"
+			"     0: do not sort the points, or\n"
+			"     1: sort the points (default).\n");
+		return 1;
+	}
+	double w;
+	int method;
+	sscanf(argv[1],"%d",&method);
+	int nupts_distribute;
+	sscanf(argv[2],"%d",&nupts_distribute);
+	sscanf(argv[3],"%lf",&w); nf1 = (int)w;  // so can read 1e6 right!
+
+	N1 = (int) nf1/upsampfac;
+	M = N1;// let density always be 1
+	if(argc>4){
+		sscanf(argv[4],"%lf",&w); M  = (int)w;  // so can read 1e6 right!
+		if(M == 0) M=N1;
+	}
+
+	FLT tol=1e-6;
+	if(argc>5){
+		sscanf(argv[5],"%lf",&w); tol  = (FLT)w;  // so can read 1e6 right!
+	}
+
+	int kerevalmeth=0;
+	if(argc>6){
+		sscanf(argv[6],"%d",&kerevalmeth);
+	}
+
+	int sort=1;
+	if(argc>7){
+		sscanf(argv[7],"%d",&sort);
+	}
+
+	int ier;
+	cout<<scientific<<setprecision(3);
+
+	FLT *x;
+	CPX *c, *fw;
+	cudaMallocHost(&x, M*sizeof(FLT));
+	cudaMallocHost(&c, M*sizeof(CPX));
+	cudaMallocHost(&fw,nf1*sizeof(CPX));
+
+	FLT *d_x;
+	CUCPX *d_c, *d_fw;
+	checkCudaErrors(cudaMalloc(&d_x,M*sizeof(FLT)));
+	checkCudaErrors(cudaMalloc(&d_c,M*sizeof(CUCPX)));
+	checkCudaErrors(cudaMalloc(&d_fw,nf1*sizeof(CUCPX)));
+
+	int dim=1;
+	CUFINUFFT_PLAN dplan = new CUFINUFFT_PLAN_S;
```

Comment:
I assume we can't call `cufinufft_makeplan` here?

Reply:
`cufinufft_makeplan` takes input of number of modes and for spread/interp only testing, we only care about the upsampled fine grid size. And it also includes e.g. making cuFFT plans, which is not needed for the test. I would think it is cleaner to not call it.

Reply:
Ok, but this suggests that maybe we're putting too much information into `cufinufft_plan`. Maybe spread/interp should have its own plan interface that can be used independent of the NUFFT call. This is nothing to resolve here, but might be worth discussing.